chainer/datasets/image_dataset.py

```diff
@@ -23,6 +26,13 @@ def _read_image_as_array(path, dtype):
     return image
 
 
+def _postprocess_image(image):
+    if image.ndim == 2:
+        # image is greyscale
+        image = image[:, :, numpy.newaxis]
```

Comment:
How about `return image[None, ...]` ?

Reply:
I guess you mean `image[..., None]`?
I confirmed it's much faster. Thanks!

Reply:
I proposed merge two operation, insert new axis and transpose.